disasm/disasm.cc

```diff
@@ -507,34 +507,52 @@ static void NOINLINE add_b1type_insn(disassembler_t* d, const char* name, uint32
   d->add_insn(new disasm_insn_t(name, match, mask | mask_rs2, {&xrs1, &branch_target}));
 }
 
-static void NOINLINE add_frtype_insn(disassembler_t* d, const char* name, uint32_t match, uint32_t mask)
+static void NOINLINE add_frtype_insn(disassembler_t* d, const char* name, uint32_t match, uint32_t mask, bool zfinx)
 {
-  d->add_insn(new disasm_insn_t(name, match, mask, {&frd, &frs1, &frs2}));
+  if (zfinx)
+    d->add_insn(new disasm_insn_t(name, match, mask, {&xrd, &xrs1, &xrs2}));
+  else
+    d->add_insn(new disasm_insn_t(name, match, mask, {&frd, &frs1, &frs2}));
```

Comment:
Would it be possible to have Zfinx redefine frd, frs1, frs2 variables, so we could avoid all these `if (zfinx)` with duplicated calls to `d->add_insn()`?

Reply:
Globals are not an acceptable alternative.

Reply:
Or we change the code into:
```
  d->add_insn(zfinx? new disasm_insn_t(name, match, mask, {&xrd, &xrs1, &xrs2}) : 
                     new disasm_insn_t(name, match, mask, {&frd, &frs1, &frs2}));
```

Reply:
I'm speaking abstractly, of course.  Think about other approaches along similar lines.